presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -448,6 +449,88 @@ public static Slice trim(@SqlType("varchar(x)") Slice slice)
         return SliceUtf8.trim(slice);
     }
 
+    @Description("remove the longest string containing only given characters from the beginning of a string")
+    @ScalarFunction("ltrim")
+    @LiteralParameters("x")
+    @SqlType("varchar(x)")
+    public static Slice leftTrim(@SqlType("varchar(x)") Slice slice, @SqlType(StandardTypes.VARCHAR) Slice charactersToTrim)
+    {
+        int[] codePointsToTrim = toCodePoints(charactersToTrim);
+        int start = leftTrimPosition(slice, codePointsToTrim);
+        return slice.slice(start, slice.length() - start);
+    }
+
+    private static int leftTrimPosition(@SqlType("varchar(x)") Slice slice, int[] codePointsToTrim)
```

Comment:
I think we do not want to annotate internal implementation with @SqlType annotation.